elephant/causality/granger.py

```diff
@@ -0,0 +1,395 @@
+
+    The blocks of size (d \times d) are set as follows:
+
+    M_ij = C(j-i)^T
+
+    where 1 \leq i \leq j \leq p. The other entries are determined by symmetry.
+
+    """
+
+    lag_covariances = _lag_covariances(data, dimension, order)
+
+    yule_walker_matrix = np.zeros((dimension*order, dimension*order))
+
+    for block_row in range(order):
+        for block_column in range(block_row, order):
+            yule_walker_matrix[block_row*dimension: (block_row+1)*dimension,
+                               block_column*dimension:
+                               (block_column+1)*dimension] = \
+                lag_covariances[block_column-block_row].T
+
+            yule_walker_matrix[block_column*dimension:
+                               (block_column+1)*dimension,
+                               block_row*dimension:
+                               (block_row+1)*dimension] = \
+                lag_covariances[block_column-block_row]
+    return yule_walker_matrix, lag_covariances
+
+
+def _vector_arm(signals, dimension, order):
+    r"""
+    Determine coefficients of autoregressive model from time series data
+
+    Parameters
+    ----------
+    signals : np.ndarray
+        time series data
+    order : int
+        order of the autoregressive model
+
+    Returns
+    -------
+    coeffs: np.ndarray
+        coefficients of the autoregressive model
+        ry
+    covar_mat : np.ndarray
+        covariance matrix of
+
+    Coefficients of autoregressive model calculated via solving the linear
+    equation
+
+    M A = C
+
+    where
+    M: Yule-Waler Matrix
+    A: Coefficients of autoregressive model
+    C: Time-shifted covariances with positive lags
+
+    Covariance matrix C_0 is then given by
+
+    C_0 = C[0] - \sum_{i=0}^{p-1} A[i]C[i+1]
+
+    where p is the orde of the autoregressive model.
+
+    """
+
+    yule_walker_matrix, lag_covariances = \
+        _yule_walker_matrix(signals, dimension, order)
+
+    positive_lag_covariances = np.reshape(lag_covariances[1:],
+                                          (dimension*order, dimension))
+
+    lstsq_coeffs = \
+        np.linalg.lstsq(yule_walker_matrix, positive_lag_covariances)[0]
+
+    coeffs = []
+    for index in range(order):
+        coeffs.append(lstsq_coeffs[index*dimension:(index+1)*dimension, ].T)
+
+    coeffs = np.stack(coeffs)
+
+    cov_matrix = np.copy(lag_covariances[0])
+    for i in range(order):
+        cov_matrix -= np.matmul(coeffs[i], lag_covariances[i+1])
+
+    return coeffs, cov_matrix
+
+
+def _optimal_vector_arm(signals, dimension, max_order,
+                        information_criterion=aic):
+    """
+    Determine optimal auto regressive model by choosing optimal order via
+    Information Criterion
+
+    Parameters
+    ----------
+    signals : np.ndarray
+        time series data
+    dimension : int
+        dimensionality of the data
+    max_order : int
+        maximal order to consider
+    information_criterion : string
```

Comment:
it's not a string anymore, it's a `callable`. Then, because `bic` and `aic` are special keywords that need to be rendered as the functions on readthedocs, they need to be enclosed in ticks '\`', and the doc should look like
```
information_criterion : callable
        `bic` for Bayesian information_criterion,
        `aic` for Akaike information criterion
```

Or you even better

":func:\`aic\`"

on your choice.

Reply:
Addressed in commit 5c26f0d. We chose the first option for the time being.

Reply:
Pay attention to the ticks. There are 3 types
```
`
'
"
```
To highlight a code snippet or a function we use the first type. But it's minor anyway. To clarify that this is not a string, I suggest adding a sentence that it's "A function to compute the information criterion". So consider

```
information_criterion : callable, optional
    A function to compute the information criterion:
        `bic` for Bayesian information_criterion,

        `aic` for Akaike information criterion
    Default: bic
```

Reply:
Fixed in commit 513790d.